notebooks/triton/deploy-bidaf-aks.ipynb

```diff
@@ -0,0 +1,259 @@
+    "response = urllib.request.urlopen(url)\n",
+    "\n",
+    "blob_client = BlobClient.from_blob_url(response.url)\n",
+    "\n",
+    "# get the root of the repo\n",
+    "prefix = Path(git.Repo(\".\", search_parent_directories=True).working_tree_dir)\n",
+    "\n",
+    "# model path\n",
+    "folder_path = prefix.joinpath(\"models\", \"triton\", \"bidaf-9\", \"1\")\n",
+    "model_file_path = prefix.joinpath(folder_path, \"model.onnx\")\n",
+    "\n",
+    "# save the model if it does not already exist\n",
+    "if not os.path.exists(model_file_path):\n",
+    "    os.makedirs(folder_path, exist_ok=True)\n",
+    "    with open(model_file_path, \"wb\") as my_blob:\n",
+    "        download_stream = blob_client.download_blob()\n",
+    "        my_blob.write(download_stream.readall())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Register model\n",
+    "\n",
+    "A registered model is a logical container stored in the cloud, containing all files located at `model_path`, which is associated with a version number and other metadata."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from azureml.core.model import Model\n",
+    "\n",
+    "model_path = prefix.joinpath(\"models\", \"triton\")\n",
+    "\n",
+    "model = Model.register(\n",
+    "    model_path=model_path,\n",
+    "    model_name=\"bidaf-9-example\",\n",
+    "    tags={\"area\": \"Natural language processing\", \"type\": \"Question-answering\"},\n",
+    "    description=\"Question answering from ONNX model zoo\",\n",
+    "    workspace=ws,\n",
+    "    model_framework=Model.Framework.MULTI,\n",
+    ")\n",
+    "\n",
+    "print(model.name, model.description, model.version)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy webservice\n",
+    "\n",
+    "In this case we deploy to a pre-created [AksCompute](https://docs.microsoft.com/python/api/azureml-core/azureml.core.compute.aks.akscompute?view=azure-ml-py#provisioning-configuration-agent-count-none--vm-size-none--ssl-cname-none--ssl-cert-pem-file-none--ssl-key-pem-file-none--location-none--vnet-resourcegroup-name-none--vnet-name-none--subnet-name-none--service-cidr-none--dns-service-ip-none--docker-bridge-cidr-none--cluster-purpose-none--load-balancer-type-none-) called \"aks-gpu-deploy,\" but for other options, see [our documentation](https://docs.microsoft.com/azure/machine-learning/how-to-deploy-and-where?tabs=azcli)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "outputPrepend"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "from azureml.core.webservice import AksWebservice\n",
+    "from azureml.core import Environment\n",
+    "from azureml.core.model import InferenceConfig\n",
+    "\n",
+    "service_name = \"triton-bidaf-9\"\n",
```

Comment:
add randomint to service name - see other examples - so this can run multiple notebook tests in parallel

Reply:
I can do this, but maybe we can use something guaranteed to be globally unique, like the path to the notebook? Using the same int range for generating service names will make this a flaky test

Reply:
all webservices are deleted daily, as long as the test isn't running 1000s of times per day there's a very low chance of conflict